hydrolib/core/io/dflowfm/ini/util.py

```diff
@@ -260,13 +263,29 @@ def get_from_subclass_defaults(cls: Type[BaseModel], fieldname: str, value: str)
         [type]: The field default that corresponds to the value.
     """
     for c in cls.__subclasses__():
-        default = c.__fields__.get(fieldname).default
-        if default.lower() == value.lower():
+        default = _try_get_default_value(c, fieldname, value)
+        if default is not None and default.lower() == value.lower():
             return default
-
     return value
 
 
+def _try_get_default_value(c, fieldname, value):
```

Comment:
Consider removing this method and instead do all in `get_from_subclass_defaults()` with these modifications:
```python
def get_from_subclass_defaults(cls: Type[BaseModel], fieldname: str, value: str, return_value_if_not_found: bool = True) -> str:
    """
    Gets a value that corresponds with the default field value of one of the subclasses.
    If the subclass doesn't have the specified field, it will look into its own subclasses
    recursively for the specified fieldname.

    Args:
        cls (Type[BaseModel]): The parent model type.
        fieldname (str): The field name for which retrieve the default for.
        value (str): The value to compare with.

    Returns:
        [type]: The field default that corresponds to the value.
    """
    for c in cls.__subclasses__():
        default = get_from_subclass_default(c, fieldname, value, False )
        if default is not None and default.lower() == value.lower():
            return default
    if return_value_if_not_found:
        return value
    else:
        return None
```

Reply:
This won't work :(

Let's assume we call this function in the context of `ForcingBase` and we want to see if any of its subclasses has a default value of `harmonic` for the `function` property.

In the `for` loop, we iterate over the children of `ForcingBase`. At some point we reach the point where `c` is equal to the `Harmonic` class. The first line in the `for` loop recursively calls the `get_from_subclass_defaults` function. So we loop over the children of the `Harmonic` class. But there are none. So the call to `get_from_subclass_defaults` return None and we eventually leave this function without returning the correct default value.

So what we would need above the `for` loop, is to check if the current class has a `fieldname` property with a default value of `value`. However, we have to change the name of the function, as it no longer only check children, but it also checks the base class. Furthermore, it will add code duplication again. That said, I am starting to think my previous implementation may still be a good candidate.

Reply:
We discussed this Monday, and indeed we need this. I have committed some final improvements in 0e24282.